src/InstrumentationEngine/InstructionGraph.cpp

```diff
@@ -17,6 +17,15 @@ MicrosoftInstrumentationEngine::CInstructionGraph::~CInstructionGraph()
         CCriticalSectionHolder lock(&m_cs);
 
         CComPtr<CInstruction> pInstr = (CInstruction*)m_pFirstInstruction;
+
+        m_pFirstInstruction.Release();
```

Comment:
Can you add some comments here, describing in more details why the order is important?

Reply:
This might be best described In terms of reference counts:

Initial ref counts:
2<=>2<=>2
After firstInstruction.Release
1<=>2<=>2
After first Disconnect
1<=>2

Reply:
I actually find this kind of confusing. I think the description of the PR is useful. Maybe we can just put that in?